### PR TITLE
Fix for issue #879, wrong cursor for selected resizeable blocks in song editor

### PR DESCRIPTION
**Pull request by craslaw**
This PR checks if TCOs are selected by the box selector before switching the cursor to the resize cursor at the edge of a selected block. 

Before (Issue #879):
The cursor updates to the resize cursor for one or more box-selected TCOs, but clicking and dragging results in moving all the selected TCOs.

After:
Same behavior as before, except the cursor does not change from the hand icon when hovering over the edge of a box-selected TCO.

From a UI stance, it makes sense to be able to resize one or more resizable blocks that are selected with the box tool. However, adding this code to the box select tool seems out of scope for this bug, and may be better off in a separate feature request - post refactor too.

### CODE REVIEW

**Comment by superpaik** [comment]
It works ok to me (windows 10-64bits)
Just one minor issue that I think it's not been introduced by this PR but maybe just double check just in case  Samples tracks now can be extended to the left, I don't know if that is intended or not (it doesn't happen in 1.2)

**Comment by PhysSong** [comment]
@Spekular While reviewing this PR, I found that knife mode is not working with selections. Is it intended?
If it is, this looks good to me. There are some merge conflicts due to renaming from `TrackContentObject`(TCO) to `Clip`, though.

**Comment by Spekular** [comment]
> @Spekular While reviewing this PR, I found that knife mode is not working with selections. Is it intended?

I didn't really have selections in mind when I wrote it. I'm not sure what the best UX would be for cutting multiple clips so for now I think it's fine to only cut one at a time?

File: src/gui/TrackContentObjectView.cpp [hunk 1]
```diff
@@ -448,13 +448,13 @@ void TrackContentObjectView::updateCursor(QMouseEvent * me)
 	SampleTCO * sTco = dynamic_cast<SampleTCO*>(m_tco);
 
 	// If we are at the edges, use the resize cursor
-	if ((me->x() > width() - RESIZE_GRIP_WIDTH && !me->buttons() && !m_tco->getAutoResize())
-		|| (me->x() < RESIZE_GRIP_WIDTH && !me->buttons() && sTco && !m_tco->getAutoResize()))
+	if ((!me->buttons() && !m_tco->getAutoResize() && !isSelected())
```
Comment (allejok96) [inline src/gui/TrackContentObjectView.cpp:451]: Redundant parenthesis for the first 3

**Review by allejok96 — APPROVED** [review]
Looks good, if you can resolve the merge conflicts I think we can merge this and close the bug. It's such a simple change. Thanks for your contribution, even if the response has been slow.

**Pull request merged**